From: Jakub Jirutka <jakub@jirutka.cz>
Date: Thu, 9 Jun 2016 01:15:00 +0200
Subject: [PATCH] Use our find-syslibs script instead of /sbin/ldconfig -p

Julia uses "ldconfig -p" to find system libraries. The problem is that musl's
ldconfig doesn't support the option "-p". Moreover, ldconfig relies on .pc
files that are installed only with -dev subpackages.

This workaround lies in using custom script that adapts output of scanelf to
the format produced by "ldconfig -p".

See https://github.com/JuliaLang/julia/issues/6742.

--- a/src/runtime_ccall.cpp
+++ b/src/runtime_ccall.cpp
@@ -31,7 +31,7 @@
     char *line=NULL;
     size_t sz=0;
 #if defined(__linux__)
-    FILE *ldc = popen("/sbin/ldconfig -p", "r");
+    FILE *ldc = popen("/usr/share/julia/find-syslibs", "r");
 #else
     FILE *ldc = popen("/sbin/ldconfig -r", "r");
 #endif
